python/tests/graph/test_pieces.py

```diff
@@ -0,0 +1,165 @@
+
+            img_arr_float = img_to_array(img).astype(np.float)
+            check_image_round_trip(img_arr_float)
+
+            img_arr_preproc = iv3.preprocess_input(img_to_array(img))
+            check_image_round_trip(img_arr_preproc)
+
+    def test_identity_module(self):
+        """ identity module should preserve input """
+
+        gfn = gfac.buildIdentity()
+        for _ in range(10):
+            m, n = prng.randint(10, 1000, size=2)
+            mat = prng.randn(m, n).astype(np.float32)
+            with IsolatedSession() as issn:
+                feeds, fetches = issn.importGraphFunction(gfn)
+                mat_out = issn.run(fetches[0], {feeds[0]: mat})
+
+            self.assertTrue(np.all(mat_out == mat))
+
+    def test_flattener_module(self):
+        """ flattener module should preserve input data """
+
+        gfn = gfac.buildFlattener()
+        for _ in range(10):
+            m, n = prng.randint(10, 1000, size=2)
+            mat = prng.randn(m, n).astype(np.float32)
+            with IsolatedSession() as issn:
+                feeds, fetches = issn.importGraphFunction(gfn)
+                vec_out = issn.run(fetches[0], {feeds[0]: mat})
+
+            self.assertTrue(np.all(vec_out == mat.flatten()))
+
+    def test_bare_keras_module(self):
+        """ Keras GraphFunctions should give the same result as standard Keras models """
+        img_fpaths = glob(os.path.join(_getSampleJPEGDir(), '*.jpg'))
+
+        for model_gen, preproc_fn in [(InceptionV3, iv3.preprocess_input),
+                                      (Xception, xcpt.preprocess_input),
+                                      (ResNet50, rsnt.preprocess_input)]:
+
+            keras_model = model_gen(weights="imagenet")
+            target_size = tuple(keras_model.input.shape.as_list()[1:-1])
+
+            _preproc_img_list = []
+            for fpath in img_fpaths:
+                img = load_img(fpath, target_size=target_size)
+                # WARNING: must apply expand dimensions first, or ResNet50 preprocessor fails
+                img_arr = np.expand_dims(img_to_array(img), axis=0)
+                _preproc_img_list.append(preproc_fn(img_arr))
+
+            imgs_input = np.vstack(_preproc_img_list)
+
+            preds_ref = keras_model.predict(imgs_input)
+
+            gfn_bare_keras = GraphFunction.fromKeras(keras_model)
+
+            with IsolatedSession(keras=True) as issn:
+                K.set_learning_phase(0)
+                feeds, fetches = issn.importGraphFunction(gfn_bare_keras)
+                preds_tgt = issn.run(fetches[0], {feeds[0]: imgs_input})
+
+            self.assertTrue(np.all(preds_tgt == preds_ref))
+
+    def test_pipeline(self):
+        """ Pipeline should provide correct function composition """
+        img_fpaths = glob(os.path.join(_getSampleJPEGDir(), '*.jpg'))
+
+        xcpt_model = Xception(weights="imagenet")
+        stages = [('spimage', gfac.buildSpImageConverter(SparkMode.RGB_FLOAT32)),
```

Comment:
The model is supposed to look like this 
![graph-function-from-list](https://user-images.githubusercontent.com/28275034/27061537-99313b82-4f99-11e7-9f55-d4127fbb2289.png)